src/ember-resource.js

```diff
@@ -819,6 +819,12 @@
 
       Ember.Resource.fetch(this, ajaxOptions)
         .done(function(json) {
+          if (self.get('isDestroying') || self.get('isDestroyed')) {
+            Ember.Resource.sendEvent(self, 'didFail');
+            result.reject();
+            self.fetched().reject();
+            return;
+          }
```

Comment:
so.. we feel strongly about rejecting the promise when this happens? the fetch itself succeeded, yeah?